ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -0,0 +1,215 @@
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Analysis (HasAnalysis (..)) where
+
+import           Control.Monad.Except
+import qualified Data.Aeson as Aeson
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Lazy as BL
+import           Data.Coerce (coerce)
+import           Data.Foldable (toList)
+import           Data.Functor.Identity
+import           GHC.Natural (Natural)
+
+import           Cardano.Binary (unAnnotated)
+import qualified Cardano.Crypto as Crypto
+import           Cardano.Slotting.EpochInfo
+import           Cardano.Slotting.Slot
+
+import qualified Cardano.Chain.Block as Chain
+import qualified Cardano.Chain.Genesis as Genesis
+import           Cardano.Chain.Slotting (EpochSlots (..))
+import qualified Cardano.Chain.Update as Update
+import qualified Cardano.Chain.UTxO as Chain
+
+import qualified Shelley.Spec.Ledger.BlockChain as SL
+import qualified Shelley.Spec.Ledger.PParams as SL
+import qualified Shelley.Spec.Ledger.Tx as Shelley
+
+import           Ouroboros.Consensus.Config
+import           Ouroboros.Consensus.Node.ProtocolInfo
+
+import qualified Ouroboros.Consensus.Byron.Ledger as Byron
+import           Ouroboros.Consensus.Byron.Node (protocolInfoByron)
+
+import qualified Ouroboros.Consensus.Shelley.Ledger.Block as Shelley
+import           Ouroboros.Consensus.Shelley.Node hiding (ShelleyGenesis)
+import qualified Ouroboros.Consensus.Shelley.Node as Shelley
+import           Ouroboros.Consensus.Shelley.Protocol.Crypto
+
+import qualified Ouroboros.Consensus.Cardano.Block as Cardano
+import           Ouroboros.Consensus.Cardano.Node (HardCodedTransition (..),
+                     protocolInfoCardano)
+
+{-------------------------------------------------------------------------------
+  HasAnalysis
+-------------------------------------------------------------------------------}
+
+class HasAnalysis blk where
+    countTxOutputs   :: blk -> Maybe Int
+    blockHeaderSize  :: blk -> Maybe Natural
+    blockTxsSize     :: blk -> Maybe [BL.ByteString]
```

Comment:
Return the length

Reply:
Eventually, I realized I actually use the transactions for 2 things: see numBlockTxs and blockTxsSize. That's why I return the transactions instead of just the size. (I missed this part when we reviewed). I changed the name though, from `blockTxsSize` to `blockTxs`.

Reply:
I can comment here!